driver/registry.go

```diff
@@ -139,17 +150,13 @@ func NewRegistry(c configuration.Provider) (Registry, error) {
 	}
 
 	// if dsn is memory we have to run the migrations on every start
-	if urlParts := strings.SplitN(dsn, "?", 1); len(urlParts) == 2 && strings.HasPrefix(dsn, "sqlite://") {
-		queryVals, err := url.ParseQuery(urlParts[1])
-		if err != nil {
-			return nil, errors.WithMessage(errors.WithStack(err), "unable to parse the DSN url")
-		}
-		if queryVals.Get("mode") == "memory" {
-			registry.Logger().Print("Kratos is running migrations on every startup as DSN is memory.\n")
-			registry.Logger().Print("This means your data is lost when Kratos terminates.\n")
-			if err := registry.Persister().MigrateUp(context.Background()); err != nil {
-				return nil, err
-			}
+	isSQLiteMemoryMode := IsSQLiteMemoryMode(dsn)
+
+	if isSQLiteMemoryMode {
```

Comment:
Since this isn't used anywhere else:

```suggestion
	if IsSQLiteMemoryMode(dsn) {
```

Reply:
Oh, you are right... didnt clean up properly after simplifying the func...